NeoML/src/Dnn/Layers/HingeLossLayer.cpp

```diff
@@ -23,10 +23,17 @@ namespace NeoML {
 
 ///////////////////////////////////////////////////////////////////////////////////
 // CHingeLossLayer
+
+void CHingeLossLayer::Reshape()
+{
+	CLossLayer::Reshape();
+	CheckArchitecture( inputDescs[1].GetDataType() == CT_Float, GetName(), "labels must be CT_Float" );
+}
+
 void CHingeLossLayer::BatchCalculateLossAndGradient(int batchSize, CConstFloatHandle data, int vectorSize,
 	CConstFloatHandle label, int labelSize, CFloatHandle lossValue, CFloatHandle lossGradient)
 {
-	NeoAssert(vectorSize == labelSize);
+	CheckArchitecture( labelSize == vectorSize, GetName(), "the labels dimensions should be equal to the first input dimensions" );
```

Comment:
What does `first` mean here? Because in NeoML order `batchSize` goes _prior_ to `vectorSize`. M.b. just use vector size?
+ move it to `::Reshape()`?